### ISSUE TITLE
Call UI not visible on android when answered from push notification in killed state

### ISSUE THREAD

**Issue by tuneerclixlogix** [voxeet/voxeet-uxkit-reactnative-firebase#5]
@codlab I was not available from long on this so now i am sharing the video from my android device. In this video you can observe that the screen went black for 1 sec that is because i covered the sensor which means the call is ongoing and also i was able to hear the voices . Its just the UI is not being displayed in this case. Its happening only for the killed state else the calling is working perfectly fine.

https://user-images.githubusercontent.com/72122766/113836249-5585d200-97aa-11eb-936c-960dbbb46c61.mp4

_Originally posted by @tuneerclixlogix in https://github.com/voxeet/voxeet-uxkit-reactnative-firebase/issues/2#issuecomment-814722792_

**Comment by codlab** [comment]
This issue is related to the react native package, can you test using the latest commit in the react-native repository
 ?
 
 `npm install --save git://github.com/voxeet/voxeet-uxkit-reactnative.git#5fc6ac4f0a642eaf5af7557d4b7fdb60965c43ba`

**Comment by codlab** [comment]
can you provide logs from `add logcat` as well?

**Comment by tuneerclixlogix** [comment]
With this repo i am facing issues while initializing 
![image](https://user-images.githubusercontent.com/72122766/114665633-e6b0f780-9d1a-11eb-9594-d25dc185a489.png)
And if i rebuild project then i am starting getting errors with other packages which i was not getting before
![image](https://user-images.githubusercontent.com/72122766/114666458-ea914980-9d1b-11eb-84b0-d4b6fc8f8b50.png)
I also tried invalidating cache and clean project

**Comment by codlab** [comment]
The new commit fixes this crash when running the app `npm install --save git://github.com/voxeet/voxeet-uxkit-reactnative.git#9ed0cf424ce065121dc7c304dbc1de29004235ca`

**Comment by tuneerclixlogix** [comment]
@codlab With this if I am starting the conference then also the UI doesn't shows up and I am not receiving push notifications with this.  
[logs.txt](https://github.com/voxeet/voxeet-uxkit-reactnative-firebase/files/6310468/logs.txt)

**Comment by tuneerclixlogix** [comment]
Seems like that was something wrong from my side as i sent the third parameter as true while initializing. Now, I have removed the third parameter and when i answer the call from killed state still the call UI is not shown.
Logs for that are 
[logs.txt](https://github.com/voxeet/voxeet-uxkit-reactnative-firebase/files/6310648/logs.txt)

**Comment by tuneerclixlogix** [comment]
I have noticed that, if I am receiving call from other device without making the call from my side even once since the app is opened then the UI is not shown either the app was in foreground or background or killed . 
If i start the conference the call UI is shown now i don't close the app and if someone else calls me and i answer then the UI is shown.

**Comment by codlab** [comment]
ok so you mean that with this version, you don't have any regression but the UI is not shown when accepting calls from killed state. I'll check this today using the latest version

In the meantime, could you forward me logs from your device since the above one was using the initialize(..., true) which was hiding the overlay completely (regardless of the type of answer)

**Comment by tuneerclixlogix** [comment]
> Seems like that was something wrong from my side as i sent the third parameter as true while initializing. Now, I have removed the third parameter and when i answer the call from killed state still the call UI is not shown.
> Logs for that are
> [logs.txt](https://github.com/voxeet/voxeet-uxkit-reactnative-firebase/files/6310648/logs.txt)

Yes @codlab , the logs in this comments are without the 3rd parameter in initialize method.

**Comment by codlab** [comment]
the latest develop branch fixes the issue

**Comment by tuneerclixlogix** [comment]
@codlab I can see there is only one branch for this repo i.e. master

**Comment by tuneerclixlogix** [comment]
@codlab I see you are talking about the develop branch of another repository https://github.com/voxeet/voxeet-uxkit-reactnative/tree/develop . So my question is the changes that you suggested we used in AndroidManifest file with this repository for the push notifications issue , will those work with the other repository also ?

**Comment by codlab** [comment]
the `@voxeet/react-native-voxeet-firebase` is only here to provide Firebase support to send voxeet's push notification to the SDK. It does not manage how the push notification are displayed or handled that's why for anything related to how it reacts, it depends on the https://github.com/voxeet/voxeet-uxkit-reactnative repository 👍 

The modification suggested in this repository are only here to make it possible to forward "non voxeet" push notification to other (declared) voxeet's push notification MessagingService receivers

... because Firebase teams found it ok to make only the first registered MessagingService registered to be able to handle FCM

**Comment by tuneerclixlogix** [comment]
@codlab Can you mention the steps to install from the develop branch ?
I tried with yarn add @voxeet/react-native-voxeet-conferencekit#develop but i am getting the error while building 
![image](https://user-images.githubusercontent.com/72122766/116246783-e63b4680-a787-11eb-8518-956c2a99cbd7.png)

Also i installed with version 0.2.11-beta13 but with this i am getting the same error as we got before initialize got 5 arguments, expected 4 .

**Comment by tuneerclixlogix** [comment]
@codlab I have this package version "@voxeet/react-native-voxeet-firebase": "^2.0.0"

**Comment by codlab** [comment]
If you installed the repo using the github's develop branch, you will need to reinstall it after you installed a package the regular way (yarn and npm reset the state of git:// project)

can you test again with the npm/yarn install of the react-native-voxeet-conferencekit's repository?

Regarding the dependency issue, it's due to binary sunsetting the hosting service. We are preparing the hosting of the uxkit packages somewhere else but in the meantime, you can use the following `android/build.gradle` modification:


```

allprojects {
    repositories {
        maven { url "https://dl.bintray.com/voxeet/maven/" }
        ...
    }
}
```

**Comment by tuneerclixlogix** [comment]
@codlab I followed the follow commands
yarn add @voxeet/react-native-voxeet-conferencekit#develop
yarn add @voxeet/react-native-voxeet-firebase
Added maven { url "https://dl.bintray.com/voxeet/maven/" } in build.gradle
Then i got the error of multiple classes used as i was using the following meta tags in AndroidManifest.xml
<meta-data
            android:name="voxeet_incoming_class"
            android:value="com.voxeet.notification.RNIncomingCallActivity" />
 <meta-data
            android:name="voxeet_incoming_accepted_class"
            android:value="com.app.trueflutterdating.MainActivity" />
 I think you guys might have included in the package also
 Now i am facing this error 
![image](https://user-images.githubusercontent.com/72122766/116555556-2b8f7d80-a91a-11eb-888b-326672aa523c.png)

**Comment by codlab** [comment]
You can follow the new readme, the package has been changed
https://github.com/voxeet/voxeet-uxkit-reactnative/tree/develop#android

There's a change I need to propagate due to a typo in the package name for the deprecated ones.

Note that you should remove the meta data since they are being handled automatically in the AndroidManifest.xml

**Comment by codlab** [comment]
You can find the mentioned fix in the latest develop commit

**Comment by tuneerclixlogix** [comment]
@codlab Yes it worked after that but still the UI is not coming up on android with the develop branch.

**Comment by tuneerclixlogix** [comment]
Here is the log file when i receive and answer the call, I am not receiving push notifications when the app is killed
[logsvoxeet.txt](https://github.com/voxeet/voxeet-uxkit-reactnative-firebase/files/6400513/logsvoxeet.txt)

**Comment by codlab** [comment]
I think I have found the reason why the notification is not appearing, can you modify your .MainActivity's launchMode to "singleTask" ? It should fix the missing popup asking for the conference following the invitation
